src/lint/concept_exercises.nim

```diff
@@ -0,0 +1,37 @@
+import std/[json, os, terminal]
+import ".."/helpers
+import "."/validators
+
+proc isValidAuthorOrContributor(data: JsonNode, key: string, path: string): bool =
+  checkObject(key)
+  checkStringKey("github_username", isRequired = true)
+  checkStringKey("exercism_username", isRequired = false)
+
+template isValidFiles(data: JsonNode, context, path: string) =
+  checkObject(context)
+  checkArrayOfStrings(context, "solution", isRequired = true)
+  checkArrayOfStrings(context, "test", isRequired = true)
+  checkArrayOfStrings(context, "exemplar", isRequired = true)
+
+proc isValidConceptExerciseConfig(data: JsonNode, path: string): bool =
+  checkObject("root")
+  checkArrayOf("authors", isValidAuthorOrContributor, isRequired = true)
+  checkArrayOf("contributors", isValidAuthorOrContributor, isRequired = false)
+  checkArrayOfStrings("", "forked_from", isRequired = false)
+  isValidFiles(data, "files", path)
+  checkStringKey("language_versions", isRequired = false)
+
+proc isEveryConceptExerciseConfigValid*(trackDir: string): bool =
+  let conceptExercisesDir = trackDir / "exercises" / "concept"
+  result = true
+  if dirExists(conceptExercisesDir):
+    for exerciseDir in getSortedSubdirs(conceptExercisesDir):
+      let configPath = exerciseDir / ".meta" / "config.json"
+      let j =
+        try:
+          parseFile(configPath)
+        except:
+          writeError("JSON parsing error", getCurrentExceptionMsg())
+          continue
```

Comment:
We can probably extract this to a helper function too as we'll have similar code for other files too.

Reply:
Agreed - let's do it after this PR. We can refactor this code to use the helper function too:
https://github.com/exercism/configlet/blob/4849007828191fb7582fdbe5275cd35beaef0fde/src/lint/lint.nim#L10-L17

This code should be improved later too - it currently assumes that any exception raised during `parseFile` is a parsing error, even if it's an `IOError`.